packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/migrations/transform_migration_version.ts

```diff
@@ -6,14 +6,12 @@
  * Side Public License, v 1.
  */
 
-import { TransformFn } from '../types';
+import { SavedObjectMigration } from '@kbn/core-saved-objects-server';
 
-export const transformMigrationVersion: TransformFn = ({ migrationVersion, ...doc }) => {
-  return {
-    transformedDoc: {
-      ...doc,
-      ...(migrationVersion ? { typeMigrationVersion: migrationVersion[doc.type] ?? '' } : {}),
-    },
-    additionalDocs: [],
-  };
+export const transformMigrationVersion: SavedObjectMigration = {
```

Comment:
The `SavedObjectMigrationParams` is too implementation-specific and only exported because our documentation-generating tooling requires so. I would rather stick to the generic `SavedObjectMigration` and adapt that in the tests.